RELEASE.md

```diff
@@ -1,23 +1,233 @@
+In `pyproject.toml` there is the `version` field - we do update it but I don't think it affects anything.
+In `docs/conf.py` there are `TAGS`, `BRANCHES`, `LATEST_VERSION`, `UNSTABLE_VERSIONS` and `DEPRECATED_VERSIONS` fields.
+
+`TAGS` and `BRANCHES` are lists of tags and branches for which documentation book will be built - those 2 lists add up to a list of versions you see when you click "expand" a version list in the left upper corner of [documentation book](https://rust-driver.docs.scylladb.com/stable/).
+
+`LATEST_VERSION` selects the element from the above list that will be the default ("stable") version shown when visiting https://rust-driver.docs.scylladb.com/stable/index.html.
+
+`UNSTABLE_VERSIONS` is a subset of `TAGS + BRANCHES`. If a version is on this list it will have a note in rendered documentation informing about instability:
+
+![The warning about unstable version](assets/unstable.png)
+
+`DEPRECATED_VERSIONS` is a subset of `TAGS + BRANCHES`. If a version is on that list then it will have a simiar note informing about deprecation:
+
+![The warning about deprecated version](assets/deprecated.png)
+
+IMPORTANT: `docs/source/conf.py` for rendering documentation is always taken from the `main` branch - even if a push to `branch-*` triggered the job.
+If that was not the case we would have to always synchronize `conf.py` on all supported branches - right now we only need to have it correct on `main` branch.
+
+It however means that any release requires updating `conf.py` on `main` branch to have the new version rendered on documentation book.
+
+## Branches and tags
+
+We maintain a branch per major release (after releasing 1.0 we may change to a branch per minor release - this is yet to be decided). This is done so that we are able to release an update for older major release (e.g. release 0.13.3 after 0.14.0 is released).
+Major version branches are named `branch-0.A.x`, where `A` is a number of the version. Example: branch for `0.14` will be `branch-0.14.x`.
+
+All new development lands on `main` branch, and may be backported by maintainers to version branches.
+
+Each release of `scylla` crate has a corresponding git tag. Tags are named `vA.B.C`, so release `0.14.1` will have tag `v0.14.1`.
+Each such tag has a message of the form `Release 0.X.Y` (e.g. `Release 0.14.0`).
+
+Below is a simplified view of how this looks.
+Diagrams created using [mermaid.js](https://mermaid.js.org/). If you view this file on GitHub it should be rendered correctly.
+You can also view it using [mermaid.live](https://mermaid.live)
+
+```mermaid
+---
+title: Rust Driver git branches
+---
+gitGraph
+   commit id: "Feature A"
+   commit id: "Release 0.13" tag: "v0.13.0"
+   branch branch-0.13.x
+   checkout branch-0.13.x
+   commit id: "Backport fix C"
+   commit id: "Release 0.13.1" tag: "v0.13.1"
+   commit id: "Backport fix D"
```

Comment:
I think `Release 0.13.2` is missing after the fix D is backported, because at the same time `Release 0.14.1` is done with fix D and this way we can misleadingly suggest that some fixes on some branches may not be released (potentially never).

Reply:
Thanks for noticing, I apprently pasted a wrong version of the graph.